INN/Largo
# 1775 largo top term refactor

## Changes

This pull request makes the following changes:

- Refactors largo_top_term to break out the getting / fetching with the output

## Why

My approach in the getting was to get a term object - that then has what we need for later use.

If a term_object is found then it is used to display the relevant term information for the top term and might be useful for #1776 
If 'none' is returned the code falls through to where that is checked and existing behavior is maintained
If an error is found then just like an empty top term setting the categories and other post types are checked to try and return something

Existing behavior is maintained - return or echo the chosen top term or a category/tag/other post type. Or if 'none' is set then return or echo an empty string

For #1775 

## Testing/Questions

Features that this PR affects:

- Might affect / benefit #1776 

Questions that need to be answered before merging:


Steps to test this PR:

1. Create test content each with one of an unset top term, top term set to a category, tag or series
2. Review if the top term is displayed for each content type
3. Change one or many top term settings for a piece of content and refresh

## Additional information

INN Member/Labs Client requesting: (if applicable)
